nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -51,8 +55,16 @@ public boolean accept(Class<?> clazz, Type genericType, MediaType mediatype) {
     }
 
     @Override
-    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) throws IOException {
-        write(jg, enrichable.getEntity());
+    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
+        try {
+            write(jg, enrichable.getEntity());
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
```

Comment:
If you receive an `InterruptedException` you must stop processing immediately, so `throw new RuntimeException("interrupted", e)`. Again, please read the doc https://doc.nuxeo.com/corg/catching-exceptions/#calling-something-declared-as-throws-exception

Reply:
I read it but then we defeat the purpose of the [ticket](https://jira.nuxeo.com/browse/NXP-28560) which is for the response not to fail because of an enricher failure.